shardingsphere-features/shardingsphere-sharding/shardingsphere-sharding-core/src/main/java/org/apache/shardingsphere/sharding/algorithm/sharding/inline/InlineShardingAlgorithm.java

```diff
@@ -84,6 +85,14 @@ public String doSharding(final Collection<String> availableTargetNames, final Pr
         return result;
     }
     
+    private String getTargetShardingNode(final Closure<?> closure, final String columnName) {
+        try {
+            return closure.call().toString();
+        } catch (final NullPointerException ex) {
```

Comment:
Is there any way to avoid cache NPE?

Reply:
I can't find other way to avoid NPE. it cannot be judged before the call() method.